apollo-rx3-support/src/main/java/com/apollographql/apollo/rx3/Rx3Apollo.java

```diff
@@ -0,0 +1,250 @@
+package com.apollographql.apollo.rx3;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+import com.apollographql.apollo.ApolloQueryWatcher;
+import com.apollographql.apollo.ApolloSubscriptionCall;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.cache.normalized.ApolloStoreOperation;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.internal.subscription.ApolloSubscriptionTerminatedException;
+import com.apollographql.apollo.internal.util.Cancelable;
+import org.jetbrains.annotations.NotNull;
+import io.reactivex.rxjava3.annotations.CheckReturnValue;
+import io.reactivex.rxjava3.core.BackpressureStrategy;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.CompletableEmitter;
+import io.reactivex.rxjava3.core.CompletableOnSubscribe;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.FlowableEmitter;
+import io.reactivex.rxjava3.core.FlowableOnSubscribe;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.ObservableEmitter;
+import io.reactivex.rxjava3.core.ObservableOnSubscribe;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.core.SingleEmitter;
+import io.reactivex.rxjava3.core.SingleOnSubscribe;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.exceptions.Exceptions;
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+
+/**
+ * The Rx3Apollo class provides methods for converting ApolloCall, ApolloPrefetch and ApolloWatcher types to RxJava 3
+ * sources.
+ */
+public class Rx3Apollo {
```

Comment:
Now that we have Kotlin in the repository, I think it would be much better to have this implemented fully in Kotlin.

That means, maybe we should merge it with RxJavaExtensions below and have everything as top-level functions.

Let me know what you think.

Reply:
Totally agree! I would love to follow up with another PR doing this - I think adding that to this PR would be a little bit out of scope since it would be done for both Rx2 and Rx3 support.

Reply:
> That means, maybe we should merge it with RxJavaExtensions below and have everything as top-level functions.

Does that mean RxJava3 support will not be usable for Java callers ?

Reply:
Ah no, the changes could be made in a way that makes it fully compatible and callable from Java.

Reply:
Yep, that can be done via JvmStatic and JvmName annotations

Reply:
@tasomaniac Would you prefer me to change this PR to include those changes as well? Or can I do a follow up PR?

Reply:
If you want to do a follow-up that would be awesome. Let's merge this then.